### DISCUSSION TITLE
Consent should be needed for analytics integration

### DISCUSSION THREAD

**Discussion by kappa-wingman** [requarks/wiki#3993] [General]
As far as I know, consent (and IP anonymization) should be needed for users in Europe, for the GDPR.
[This link](https://consent.guide/google-analytics-cookie-consent/) provides some thought about the Google Analytics and consent.

So the question is how to enable consent to users for all those analytics products (Google Analytics, Hotjar, etc.)

**Comment by Wunderharke** [comment]
Is there any progress on this? To operate a public instance in Europe this is crucial.

**Comment by ayoahha** [comment]
Agree, i have the same question/interrogation...
**How can we add some GDPR information for users ?**

**Comment by patrickdung** [comment]
For my wiki site, I add the cookie consent in the Theme, HTML head injection.
It loaded the cookie consent CSS and JS, and display a consent message at the bottom of the screen.'
It just display the consent and has no integration with any of the analytics applications supported by Wikijs.

Of course if the cookie consent is configurable in the WikiJS UI, it would be great.

**Reply by ayoahha** [reply]
Thanks for sharing ! Good idea indeed 
Will try this... Is there any text standard ?

**Reply by patrickdung** [reply]
Here's an example. The code / formatting is for your reference.
You are responsible for the cookie consent and compliance with GDPR. 
I am not a lawyer, use it at your own risk!

```
<link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/cookieconsent2/3.1.1/cookieconsent.min.css" integrity="sha384-nA7usz4cMKo7RylsCZnpe0V9qx5X7sa011o0OrDzrKUE7Ero1gB48Fjaw6SFxVZe" crossorigin="anonymous"/>
<script async defer src="https://cdnjs.cloudflare.com/ajax/libs/cookieconsent2/3.1.1/cookieconsent.min.js" integrity="sha384-xLPjH6nqG4p/2mu1KJ1t+4KqcvkyXVwBucQtyGK65lDNfKzECxGY/QEYWR+Wompr" crossorigin="anonymous">
</script>

<script>
window.addEventListener("load", function(){
 window.cookieconsent.initialise({
  "palette": {
    "popup": {
      "background": "#252525",
      "text": "#f0f0f0"
    },
    "button": {
      "background": "#ff6600"
    }
  },
  "theme": "edgeless",
  "cookie": {
        "secure": true
  },
  "content": {
    "message": "Please read the privacy policy and allow cookies.",
    "allow": "Agree",
    "dismiss": "Acknowledged",
    "link": "Read more",
    "href": "https://www.cloudnative.quest/pages/privacy-policy/"
  }
 })
});

</script>
```

FYR, here's my [wikijs](https://wikijs.cloudnative.quest)

**Reply by ayoahha** [reply]
Many thanks :+1:
